wrappers/csharp/cs-tutorial-8-D400-on-chip-calibration/ExampleAutocalibrateDevice.cs

```diff
@@ -0,0 +1,541 @@
+                                Console.WriteLine($"Calibration aborted");
+                                Console.WriteLine($"Stopping calibration pipeline...");
+                                pipeline.Stop();
+                                return;
+                            }
+                        }
+
+                        if (!thisCalibrationfault)
+                        {
+                            Console.WriteLine($"\n\t Time spend: {sw.Elapsed.ToString(@"mm\:ss\.fff")}  (min:sec.millisec)");
+                            Console.WriteLine($"\t Device health: {health} ({_deviceHealthDescription[GetDeviceHealth(health)]})");
+
+                            var res = ConsoleGetKey(new[] { ConsoleKey.Y, ConsoleKey.N, ConsoleKey.A }, @"Accept calibration ? Yes/No/Abort");
+                            Console.WriteLine("User's selection = {0}", res);
+                            if (res == ConsoleKey.A)
+                                abortRequested = true;
+                            else
+                                succeedOnChipCalibration = (res == ConsoleKey.Y);
+                        }
+                    }
+
+                    Console.WriteLine("Step 2 :  done");
+
+                    // 3. Toggle between calibration tables to assess which is better. This is optional.   
+                    Console.WriteLine($"{Environment.NewLine}3. Toggle between calibration tables to assess which is better.");
+                    aCalibratedDevice.CalibrationTable = calTableAfter;
+                    Console.WriteLine("Step 3 :  done");
+
+                    // 4. burns the new calibration to FW persistently. 
+                    Console.WriteLine("");
+                    if (ConsoleKey.Y == ConsoleGetKey(new[] { ConsoleKey.Y, ConsoleKey.N },
+                                                      @"4. Burns the new calibration to FW persistently. (Y\N)"))
+                    {
+                        aCalibratedDevice.WriteCalibration();
+                        Console.WriteLine("Step 4 :  done");
+                    }
+                    else
+                        Console.WriteLine("Step 4 :  skipped");
+
+                    Console.WriteLine("Calibration complete");
+                    Console.WriteLine($"{Environment.NewLine}Stopping calibration pipeline...");
+
+                    pipeline.Stop();
+                }
+
+                //}
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"{Environment.NewLine} Error during calibration:{Environment.NewLine} {ex.Message}");
+
+            }
+
+        }
+
+
+        #region TestCalibrationModes
+        public void TestCalibrationModes(Device dev = null)
+        {
+
+            try
+            {
+
+                if (dev == null)
+                {
+                    Console.WriteLine($"{Environment.NewLine}Getting context...");
+                    dev = new Context().QueryDevices().First();
```

Comment:
check dev not null after this line

Reply:
The condition is checked 4 lines further